x/claim/keeper/activity.go

```diff
@@ -0,0 +1,89 @@
+package keeper
+
+import (
+	"encoding/json"
+
+	"github.com/c-osmosis/osmosis/x/claim/types"
+	"github.com/cosmos/cosmos-sdk/store/prefix"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+// GetActivities get activites of users for genesis export
+func (k Keeper) GetActivities(ctx sdk.Context) []types.UserActivities {
+	store := ctx.KVStore(k.storeKey)
+	prefixStore := prefix.NewStore(store, append([]byte(types.ActionKey)))
+	iterator := prefixStore.Iterator(nil, nil)
+	defer iterator.Close()
+
+	actionsByUser := make(map[string]types.Actions)
+	for ; iterator.Valid(); iterator.Next() {
+		value := types.UserActivity{}
+		err := json.Unmarshal(iterator.Value(), &value)
+		if err != nil {
+			panic(err)
+		}
+		actions := actionsByUser[value.User]
+		actions = append(actions, value.Action)
+		actionsByUser[value.User] = actions
+	}
+
+	activities := []types.UserActivities{}
+	for user, actions := range actionsByUser {
+		activities = append(activities, types.UserActivities{
+			User:      user,
+			Actions:   actions,
+		})
+	}
+	return activities
+}
+
+func (k Keeper) SetUserActions(ctx sdk.Context, address sdk.AccAddress, actions []types.Action) {
```

Comment:
I'm a bit confused by the distinction between userActions and UserActivities

Reply:
I think yes, will update to UserActions - same meaning